### PR TITLE
update the docstring for `QNGOptimizer`

### LINKED ISSUE #3129: [BUG] The documentation for the QNG Optimizer includes a deprecated item.
### Expected behavior

The [documentation for the QNG Optimizer](https://docs.pennylane.ai/en/stable/code/api/pennylane.QNGOptimizer.html) doesn't use deprecated functions.

### Actual behavior

The [documentation for the QNG Optimizer](https://docs.pennylane.ai/en/stable/code/api/pennylane.QNGOptimizer.html) includes a reference to ExpvalCost in the first example. This should be changed for the new way of using the optimizer since expval doesn't work as-is for the example.

### Additional information

_No response_

### Source code

```shell
import pennylane as qml

dev = qml.device("default.qubit", wires=1)
def circuit(params, wires=0):
    qml.RX(params[0], wires=wires)
    qml.RY(params[1], wires=wires)
coeffs = [1, 1]
obs = [qml.PauliX(0), qml.PauliZ(0)]
H = qml.Hamiltonian(coeffs, obs)
cost_fn = qml.ExpvalCost(circuit, H, dev)
```

### Tracebacks

```shell
/opt/conda/envs/pennylane/lib/python3.9/site-packages/pennylane/vqe/vqe.py:154: UserWarning: ExpvalCost is deprecated, use qml.expval() instead. For optimizing Hamiltonian measurements with measuring commuting terms in parallel, use the grouping_type keyword in qml.Hamiltonian.
  warnings.warn(
```

### System information

```shell
Name: PennyLane
Version: 0.25.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /opt/conda/envs/pennylane/lib/python3.9/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Cirq, PennyLane-Lightning, PennyLane-Lightning-GPU, PennyLane-qiskit, pennylane-qulacs, PennyLane-SF

Platform info:           Linux-5.4.196-108.356.amzn2.x86_64-x86_64-with-glibc2.31
Python version:          3.9.13
Numpy version:           1.23.2
Scipy version:           1.9.0
Installed devices:
- lightning.gpu (PennyLane-Lightning-GPU-0.24.1)
- default.gaussian (PennyLane-0.25.0)
- default.mixed (PennyLane-0.25.0)
- default.qubit (PennyLane-0.25.0)
- default.qubit.autograd (PennyLane-0.25.0)
- default.qubit.jax (PennyLane-0.25.0)
- default.qubit.tf (PennyLane-0.25.0)
- default.qubit.torch (PennyLane-0.25.0)
- default.qutrit (PennyLane-0.25.0)
- lightning.qubit (PennyLane-Lightning-0.25.0)
- cirq.mixedsimulator (PennyLane-Cirq-0.24.0)
- cirq.pasqal (PennyLane-Cirq-0.24.0)
- cirq.qsim (PennyLane-Cirq-0.24.0)
- cirq.qsimh (PennyLane-Cirq-0.24.0)
- cirq.simulator (PennyLane-Cirq-0.24.0)
- qiskit.aer (PennyLane-qiskit-0.24.0)
- qiskit.basicaer (PennyLane-qiskit-0.24.0)
- qiskit.ibmq (PennyLane-qiskit-0.24.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.24.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.24.0)
- qulacs.simulator (pennylane-qulacs-0.24.0)
- strawberryfields.fock (PennyLane-SF-0.20.1)
- strawberryfields.gaussian (PennyLane-SF-0.20.1)
- strawberryfields.gbs (PennyLane-SF-0.20.1)
- strawberryfields.remote (PennyLane-SF-0.20.1)
- strawberryfields.tf (PennyLane-SF-0.20.1)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by albi3ro**
Fixes #3129 

The docstring for `QNGOptimizer` was still using `ExpValCost` and `QNodeCollection`s, even though those objects are both deprecated.  This fixes that.

### CODE REVIEW

File: pennylane/optimize/qng.py [hunk 1]
```diff
@@ -73,68 +73,35 @@ class QNGOptimizer(GradientDescentOptimizer):
...
-    >>> theta_new = opt.step(cost_fn, init_params)
-    >>> print(theta_new)
-    [0.011445239214543481, -0.027519522461477233]
+    >>> theta_new = opt.step(circuit, init_params)
+    >>> theta_new
+    tensor([ 0.01100528, -0.02799954], requires_grad=True)
 
-    Alternatively, the same objective function can be used for the optimizer
-    by manually providing the ``metric_tensor_fn``.
+    An alternative way to calculate the metric tensor can be provided to :meth:`~.step`
```
Comment (josh146) [inline pennylane/optimize/qng.py:98]:
```suggestion
    An alternative function to calculate the metric tensor of the QNode
    can be provided to :meth:`~.step`
```

File: pennylane/optimize/qng.py [hunk 2]
```diff
@@ -73,68 +73,35 @@ class QNGOptimizer(GradientDescentOptimizer):
...
-        For the following cases, providing metric_tensor_fn may be useful:
-
-        * For hybrid classical-quantum models, the "mixed geometry" of the model
-          makes it unclear which metric should be used for which parameter.
-          For example, parameters of quantum nodes are better suited to
-          one metric (such as the QNG), whereas others (e.g., parameters of classical nodes)
-          are likely better suited to another metric.
-
-        * For multi-QNode models, we don't know what geometry is appropriate
-          if a parameter is shared amongst several QNodes.
```
Comment (josh146) [inline pennylane/optimize/qng.py:91]: @albi3ro I suggest keeping this part (up to line 91), with the following change:

- Changing the first sentence to say "The QNG optimizer by default only supports single QNodes as objective functions." (that is, remove mention of `ExpvalCost`).

- Maybe moving it after the example?

The reason is that a lot of users expect `QNGOptimizer` to work on arbitrary cost functions, but this is not the case, for theoretical reasons --- it is not clear how to generalize the theory beyond a single variational circuit. We used to get a lot of questions/bug reports about this, so decided to be explicit in the docstring.

**Review by josh146 — APPROVED** [review]
Thanks @albi3ro! Looks good from my end, but strongly suggest keeping the first part of the note.

File: pennylane/optimize/qng.py [hunk 3]
```diff
@@ -75,11 +75,11 @@ class QNGOptimizer(GradientDescentOptimizer):
 
     .. note::
 
-        The QNG optimizer supports single QNodes or :class:`~.ExpvalCost` objects as objective functions.
-        Alternatively, the metric tensor can directly be provided to the :func:`step` method of the optimizer,
-        using the ``metric_tensor_fn`` argument.
+        The QNG optimizer supports single QNodes objects as objective functions. Alternatively,
```
Comment (albi3ro) [inline pennylane/optimize/qng.py:78]:
```suggestion
        The QNG optimizer supports using a single :class:`~.QNode` as the objective function. Alternatively,
```

**Pull request merged**